Cargo.toml

```diff
@@ -243,6 +244,9 @@ features = [
   "Win32_System_Threading",
 ]
 
+[target.'cfg(not(target_env = "msvc"))'.dependencies]
```

Comment:
Should we restrict this to only linux and macos ?

Reply:
Maybe? Most other sibling projects test only on macOS and Linux for it.

* https://github.com/rust-lang/rust/blob/9322d183f45e0fd5a509820874cc5ff27744a479/config.example.toml?plain=1#L723-L725
* https://github.com/rust-lang/rust/blob/9322d183f45e0fd5a509820874cc5ff27744a479/src/tools/miri/Cargo.toml?plain=1#L31-L36

But rust-analyzer doesn't.

* https://github.com/rust-lang/rust/blob/9322d183f45e0fd5a509820874cc5ff27744a479/src/tools/rust-analyzer/crates/rust-analyzer/Cargo.toml?plain=1#L77-L78
* https://github.com/rust-lang/rust-analyzer/blob/418c1365eccf20c9261b6948a6e637f789224af9/crates/rust-analyzer/src/bin/main.rs#L31-L33

Reply:
CI failed for `Windows x86_64 gnu`, so we may want to disable it for Windows.

Reply:
Solved by switching to `mimalloc` instead.